### PR TITLE
INTMDB-275: Changed the pointer in some paremeters for custom db role

### PR DESCRIPTION
**Pull request by coderGo93**
## Description

- Updated vendor and changed to pointer string in parameter `DB`, `Collection` of Custom DB roles resource

Link to any related issue(s):

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the Terraform contribution guidelines
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code

### CODE REVIEW

**Review by thetonymaster — APPROVED** [review]
LGTM

**Review by themantissa — APPROVED** [review]
LGTM

**Review by threebee — APPROVED** [review]
LGTM

File: go.sum [hunk 1]
```diff
@@ -1199,6 +1199,10 @@ go.mongodb.org/atlas v0.13.0 h1:JkJOWsKm9k2mcFaivaaMNDpKDsxJJj1O0eUsDtnNvuE=
 go.mongodb.org/atlas v0.13.0/go.mod h1:wVCnHcm/7/IfTjEB6K8K35PLG70yGz8BdkRwX0oK9/M=
 go.mongodb.org/atlas v0.13.1-0.20211015211338-c17a67cbe2bf h1:AczRmIosmEvyLpHNnvykZlUpDv90oZ0FEf1zs+CNYH0=
 go.mongodb.org/atlas v0.13.1-0.20211015211338-c17a67cbe2bf/go.mod h1:wVCnHcm/7/IfTjEB6K8K35PLG70yGz8BdkRwX0oK9/M=
+go.mongodb.org/atlas v0.13.1-0.20211103172854-77290cf48e1e h1:Fcx4rdJT5074x9TIENhX59vaaxgkkpMPhbJKql+FqxA=
+go.mongodb.org/atlas v0.13.1-0.20211103172854-77290cf48e1e/go.mod h1:lQhRHIxc6jQHEK3/q9WLu/SdBkPj2fQYhjLGUF6Z3U8=
+go.mongodb.org/atlas v0.13.1-0.20211104090717-43497bed4998 h1:FVIc+7lT1gXYoR47uBOTq7079ZEjlTTlznDAv/zUqrk=
+go.mongodb.org/atlas v0.13.1-0.20211104090717-43497bed4998/go.mod h1:lQhRHIxc6jQHEK3/q9WLu/SdBkPj2fQYhjLGUF6Z3U8=
```
Comment (gssbzn) [inline go.sum]: I think a `go mod tidy` is missing here

**Review by gssbzn — APPROVED** [review]
LGTM we can do a `go mod tidy` as a separate PR

**Review by threebee — APPROVED** [review]
LGTM. Thanks for the changes.

**Pull request merged**